### PR TITLE
Emit error when cannot connect to kubernetes

### LINKED ISSUE #3313: Fix `linkerd install` error message when no cluster found
When my environment cannot connect to a k8s cluster, and I run `linkerd install`, I get `It appears that there is an existing installation`:

```bash
$ linkerd install
Unable to install the Linkerd control plane. It appears that there is an existing installation:

Get https://kubernetes.docker.internal:6443/version?timeout=30s: dial tcp 127.0.0.1:6443: connect: connection refused

If you are sure you'd like to have a fresh install, remove these resources with:

    linkerd install --ignore-cluster | kubectl delete -f -

Otherwise, you can use the --ignore-cluster flag to overwrite the existing global resources.
```

### PR DESCRIPTION
**Pull request by alenkacz**
Right now when cannot connect to kubernetes, error indicates that there are existing resources in the cluster. This PR introduces separate error message just for checking connectivity to kubernetes.

Fixes #3313

Signed-off-by: Alena Varkockova <varkockova.a@gmail.com>

### CODE REVIEW

**Comment by alenkacz** [comment]
@siggy :)

**Comment by alenkacz** [comment]
@siggy as discussed I went the CategoryError road, feel free to review :)

File: cli/cmd/install.go [hunk 1]
```diff
@@ -793,39 +820,38 @@ func (options *installOptions) proxyConfig() *pb.Proxy {
...
 			}
 		}
 	})
 
-	if len(errMsgs) > 0 {
-		return errors.New(strings.Join(errMsgs, "\n"))
-	}
-
-	return nil
+	return outputError
```
[the remark this answers is not in the document]
Comment (alenkacz) [reply cli/cmd/install.go]: oh wow, and I really tested it but this one I did not see coming. Anyway I implemented exactly what you propose, feel free to re-review :)

But I am starting to feel the code is getting a bit messy with the error types etc. I guess we should work at least on test coverage...

[the remark this answers is not in the document]
Comment (siggy) [reply cli/cmd/install.go]: Yeah, totally agree these subtle issues popping up are things that should be caught by tests. This PR looks great though, really appreciate it!

**Pull request merged**